tests/esp32/esp32_idf_heap_info.py

```diff
@@ -0,0 +1,24 @@
+try:
+    import esp32
+except ImportException:
+    print("SKIP")
+    raise SystemExit
+
+# try getting heap regions, looks like a couple of small regions are always taken?
+heap = esp32.idf_heap_info(esp32.MEM_DATA)
+# print("heap:", heap)
+if heap[1] == (7288, 0, 0, 0):
```

Comment:
Will these specific numbers remain stable across different esp32 builds, or IDF versions?  It won't work to update this test each time the IDF version changes, or IDF configuration changes (SDK configuration which may change initial memory layout).

With these kinds of things the tests usually just check the length of the tuple and types of entries.  Here, it should be possible to test that the first 3 entries are always `>=` the next one.

Also, tests usually just print out their results (they don't check values in the test), and the .exp file contains the correct values which are tested against.